### PR TITLE
feat(ui): support numeric custom properties in Model Registry UI

### PR DESCRIPTION
**Pull request by Taj010**
## Description
The Model Registry UI currently filters out custom properties with numeric types (`int_value`, `double_value`), displaying only string properties. While the backend correctly stores all property types, users cannot see or edit numeric metadata in the UI.

This PR expands UI support to include INT and DOUBLE property types:

1. **Display**: Numeric properties now appear in the properties table (rendered as strings)
2. **Edit**: Type detection automatically creates properties with the correct type:
   - Integer values (`/^-?\d+$/`) → INT type
   - Decimal values (`/^-?\d+\.\d+$/`) → DOUBLE type
   - Everything else → STRING type
3. **Type transitions**: Users can edit properties and change types (e.g., "42" → "42.5" converts INT → DOUBLE)

Scope
- ✅ Added support for STRING, INT, and DOUBLE types
- ✅ Labels continue to work unchanged (STRING with empty value)
-  ⏸️ Deferred STRUCT, PROTO, BOOL types (require specialized UI)

Changes
- `utils.ts`: Added `getPropertyValue()` helper and type detection in `mergeUpdatedProperty()`
- `ModelPropertiesExpandableSection.tsx`: Updated to use value extraction helper
- Added comprehensive test coverage (20+ new tests)

## How Has This Been Tested?
- New regression tests for numeric property display
- New tests for type detection (INT, DOUBLE, STRING)
- Tests for type transitions (numeric ↔ string, int ↔ double)
- Edge case coverage (negative numbers, empty strings, scientific notation)
- Existing tests pass with numeric properties included

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [X] The commits have meaningful messages
- [X] Automated tests are provided as part of the PR for major new functionalities; testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work.
- [X] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).
- [ ] **For first time contributors**: Please reach out to the [Reviewers](https://github.com/kubeflow/model-registry/blob/main/OWNERS) to ensure all tests are being run, ensuring the label `ok-to-test` has been added to the PR.

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/__tests__/utils.spec.ts [hunk 1]
```diff
@@ -313,6 +373,154 @@ describe('mergeUpdatedProperty', () => {
...
+    it('should create STRING property for values with leading zeros', () => {
+      const customProperties: ModelRegistryCustomProperties = {};
+      const result = mergeUpdatedProperty({
+        customProperties,
+        op: 'create',
+        newPair: { key: 'version', value: '007' },
+      });
+      expect(result).toEqual({
+        version: { int_value: '007', metadataType: ModelRegistryMetadataType.INT },
+      } satisfies ModelRegistryCustomProperties);
```
Comment (mturley) [inline clients/ui/frontend/src/app/pages/modelRegistry/screens/__tests__/utils.spec.ts:459]: This test name says "should create STRING property for values with leading zeros" but the assertion expects INT (`int_value: '007'`). The regex `^-?\d+$` does match `"007"`, so INT is the actual behavior.

Either:
- **Fix the test name** to match the behavior: e.g. `"should create INT property for values with leading zeros"` (if `007` → INT is intentional)
- **Or fix the behavior** to treat leading-zero strings like `"007"` as STRING (if preserving the leading zero semantics matters — `007` as an integer would normally be `7`)

I'd lean toward treating leading-zero values as STRING, since leading zeros are typically meaningful (version numbers, codes, etc.) and would be lost if round-tripped through an integer. But either way, the test name should match what's actually being asserted.

**Comment by mturley** [comment]
/lgtm
/approve
Thanks!
